web/packages/teleterm/src/ui/Search/actions.tsx

```diff
@@ -261,3 +294,17 @@ export function mapToAction(
       assertUnreachable(result);
   }
 }
+
+async function addResourceToRequest(
+  ctx: IAppContext,
+  resource: ResourceRequest
+): Promise<void> {
+  const rootClusterUri = routing.ensureRootClusterUri(resource.resource.uri);
+  const { isAtDesiredWorkspace } =
+    await ctx.workspacesService.setActiveWorkspace(rootClusterUri);
+  if (isAtDesiredWorkspace) {
+    await ctx.workspacesService
+      .getWorkspaceAccessRequestsService(rootClusterUri)
+      .addResource(resource);
+  }
+}
```

Comment:
Selecting an resource I've already added to a request sends a `GetRequestableRoles` RPC. Is there a simple way in which we could avoid that?

Reply:
At the moment, resources exist in a state of "you have access to it" and "requires request". Off the top of my head, we could maybe avoid the fetch if we do some sort of check like "if this is being added to the request but `requiresRequest` isn't explicitly `true`" (which would imply we are in "request" mode), then maybe don't fetch it. however, that could interfere with the roles required for the two resources to be available (assuming some weird interaction with roles needed for the explicitly `requiresRequest` resources, and the ones you have access to already).

Long story longer, I believe we should just call it all times to get the "most correct" list. I won't die on this hill tho

Reply:
Actually, the reason is quite simple, when adding a resource again, we overwrite it and the state changes. This in turn causes `useEffect` to fire the request. 
I added a check that skips adding a resource if it is already added.